File: algorithmByJava/src/main/java/com/github/sfyc23/algorithm/classicfifty/Example40.java
package com.github.sfyc23.algorithm.classicfifty;

import java.util.Arrays;

/**
 * Author :leilei on 2017/6/28 12:39
 * 【程序40】  题目：字符串排序。
 */
public class Example40 {
    public static void main(String[] args) {
        String s = "redistribution";
        System.out.println("原:" + s);

        char[] ca = s.toCharArray();
        Arrays.sort(ca);
        String newS = String.valueOf(ca);
        System.out.println("排序后:" + newS);
    }
}

File: algorithmBykotlin/src/main/java/com/github/sfyc23/algorithm/classicfifty/Example40.kt
package com.github.sfyc23.algorithm.classicfifty

/**
 * Author :leilei on 2017/6/28 12:42
 * 【程序40】  题目：字符串排序。
 */
fun main(args: Array<String>) {
    var s = "redistribution";
    println("原:$s")
    println("排序:${s.sort()}")
}

fun String.sort() :String {
    var ca = this.toCharArray()
    ca.sort()
    return String(ca);
}